### ISSUE TITLE
create-project ignores signals (?)

### ISSUE THREAD

**Issue by alcohol** [composer/composer#6836]
When running `composer create-project`, it seems we do not handle signals such as `SIGINT`, etc.

**Comment by alcohol** [comment]
Hmm maybe my expectations are just wrong.

```
rob@macbookpro.local /tmp/composer-test > composer create-project laravel/laravel
Installing laravel/laravel (v5.5.0)
  - Installing laravel/laravel (v5.5.0): Loading from cache
Created project in /private/tmp/composer-test/laravel
> @php -r "file_exists('.env') || copy('.env.example', '.env');"
Loading composer repositories with package information
Updating dependencies (including require-dev)
^C^C^C^C^C^C^C^C^C^C^C^C^C^CPackage operations: 68 installs, 0 updates, 0 removals
  - Installing vlucas/phpdotenv (v2.4.0): Loading from cache
  - Installing symfony/css-selector (v3.3.13): Loading from cache
  - Installing tijsverkoyen/css-to-inline-styles (2.2.0): Loading from cache
  - Installing symfony/polyfill-mbstring (v1.6.0): Loading from cache
^C  - Installing symfony/var-dumper (v3.3.13): Loading from cache
  - Installing symfony/routing (v3.3.13): Loading from cache
  - Installing symfony/process (v3.3.13): Loading from cache
  - Installing symfony/http-foundation (v3.3.13): Loading from cache
  - Installing symfony/event-dispatcher (v3.3.13): Loading from cache
  - Installing psr/log (1.0.2): Loading from cache
^C  - Installing symfony/debug (v3.3.13): Loading from cache
  - Installing symfony/http-kernel (v3.3.13): Loading from cache
  - Installing symfony/finder (v3.3.13): Loading from cache
  - Installing symfony/console (v3.3.13): Loading from cache
  - Installing doctrine/lexer (v1.0.1): Loading from cache
^C    Failed to download doctrine/lexer from dist: The process has been signaled with signal "2".
    Now trying to download from source
  - Installing doctrine/lexer (v1.0.1): ^CCloning 83893c552f
^C

  [Symfony\Component\Process\Exception\RuntimeException]
  The process has been signaled with signal "2".


create-project [-s|--stability STABILITY] [--prefer-source] [--prefer-dist] [--repository REPOSITORY] [--repository-url REPOSITORY-URL] [--dev] [--no-dev] [--no-custom-installers] [--no-scripts] [--no-progress] [--no-secure-http] [--keep-vcs] [--no-install] [--ignore-platform-reqs] [--] [<package>] [<directory>] [<version>]

```

It seems the symfony process component does capture it eventually. It just takes much longer than I would expect. Also, it seems our own handler does not trigger:

https://github.com/composer/composer/blob/bfed974ae969635e622c4844e5e69526d8459baf/src/Composer/Command/CreateProjectCommand.php#L318-L326

**Comment by stof** [comment]
@alcohol PHP is synchronous. And PHP 7 changed the behavior of `declare(ticks=100)`. so maybe this does not properly interrupt Composer anymore.

**Comment by Seldaek** [comment]
I think this is fixed in 2.0
